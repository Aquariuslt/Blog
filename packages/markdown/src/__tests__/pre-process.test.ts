import { read } from '@/__tests__/helper';
import * as matter from 'gray-matter';

describe('markdown: preprocess', () => {
  it('# should read file content in pre-process steps', () => {
    const raw = read(`sample-article.md`);
    const { data, content } = matter(raw);

    expect(data).not.toBeUndefined();
    expect(data).toHaveProperty('title');
    expect(data).toHaveProperty('created');
    expect(content).not.toBeUndefined();

    expect(data).toMatchSnapshot();
    expect(content).toMatchSnapshot();
  });
});
