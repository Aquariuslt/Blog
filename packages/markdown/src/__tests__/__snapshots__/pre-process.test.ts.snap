// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown: preprocess # should read file content in pre-process steps 1`] = `
Object {
  "category": Array [
    "Blog",
  ],
  "created": 2019-09-10T00:00:00.000Z,
  "permalink": "introducing-json-properties-loader",
  "tags": Array [
    "NPM",
    "Node",
    "Typescript",
  ],
  "title": "introducing json properties loader",
  "updated": 2019-09-10T00:00:00.000Z,
}
`;

exports[`markdown: preprocess # should read file content in pre-process steps 2`] = `
"
# A beautiful day

This is a example markdown content with all one-pass test cases.

## Getting Started

Getting started snippet

### Installing

\`\`\`shell script
yarn add -D properteis-json-loader
\`\`\`

or, using npm

\`\`\`shell script
npm install --save-dev properties-json-loader
\`\`\`

### Update webpack configuration

You should use it to load as one of webpack loader configuration matching \`*.properties\` file.

## Deep Understanding


## References


![Absolute Image](https://img.aquariuslt.com/posts/2019/08/migrating-github-actions.png)
![Relative Image](images/sample-image.png)
"
`;
